### DISCUSSION TITLE
When subclassing - why isn't _type returned on query?

### DISCUSSION THREAD

**Discussion by cmc476** [mongodb/mongoid#5155] [Q&A]
ie...
```
class Pet
  include Mongoid::Document
  field :name, type: String
end

class Dog < Pet
  field :breed, type: String
  field :average_weight, type: Integer
end

class Cat < Pet
  field :minion_strength, type: Integer
end

get '/pets' do
  Pet.all.to_json
end
```
get returns all attributes but `_type`...though it's in db.

**Comment by p-mongo** [comment]
Hi @cmc476 ,

I'm not sure why this is Mongoid's behavior, but if ActiveRecord does return its STI class value in the equivalent scenario we would consider changing Mongoid to match.

**Comment by neilshweky** [comment]
Hi there! I have created the following jira ticket to investigate your report: https://jira.mongodb.org/browse/MONGOID-5367

**Reply by neilshweky** [reply]
Hi there, after some investigation I have found that there is a Mongoid configuration option called `include_type_for_serialization` which, when set to true, will include the _type field in the results of as_json. You can set it as follows:
```
Mongoid.include_type_for_serialization = true
```
or in your `mongoid.yml`.
